extensions/amp-inline-gallery/1.0/amp-inline-gallery.js

```diff
@@ -37,13 +39,59 @@ import {userAssert} from '../../../src/log';
 const TAG = 'amp-inline-gallery';
 
 class AmpInlineGallery extends PreactBaseElement {
+  /** @param {!AmpElement} element */
+  constructor(element) {
+    super(element);
+
+    /** @private {?Element} */
+    this.carousel_ = null;
+
+    /** @private {?ResizeObserver} */
+    this.resizeObserver_ = null;
+  }
+
   /** @override */
   init() {
+    this.carousel_ = this.element.parentElement.querySelector(
+      'amp-base-carousel'
+    );
+    this.resizeObserver_ = new this.win.ResizeObserver(() => {
+      // If carousel resizes, update height variable.
+      setStyle(
+        this.element,
+        '--i-amphtml-carousel-height',
+        px(this.carousel_.offsetHeight)
```

Comment:
Should we just use `clientRect` from the resize observer's callback and avoid extra measurement entirely?

Reply:
Good point. I had originally discarded `clientRect` because it did not give the same `top` or `y` value as `offsetTop`. But when I changed it to no longer update `--i-amphtml-carousel-top`, we no longer needed that information from the `clientRect`. I'll modify.